src/paas_charm/utils.py

```diff
@@ -103,3 +103,24 @@ def config_get_with_secret(
     if secret_id is None:
         return None
     return charm.model.get_secret(id=typing.cast(str, secret_id))
+
+
+def get_missing_non_optional_configs(charm: ops.CharmBase) -> list[str] | None:
```

Comment:
What is the reason for returning `None`? Why not just return an empty collection?